src/Native/configure.cmake

```diff
@@ -47,6 +48,38 @@ check_cxx_source_compiles(
     "
     HAVE_GNU_STRERROR_R)
 
+
+if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+    set (CMAKE_REQUIRED_LIBRARIES rt)
+endif ()
+
+check_cxx_source_runs(
+    "
+    #include <sys/mman.h>
+    #include <fcntl.h>
+    #include <unistd.h>
+
+    int main()
+    {
+        int fd = shm_open(\"/corefx_configure_shm_open\", O_CREAT | O_RDWR, 0777);
+        if (fd == -1)
+            return -1;
+
+        shm_unlink(\"/corefx_configure_shm_open\");
+
+        // NOTE: PROT_EXEC, and MAP_PRIVATE don't work well ell with shm_open
```

Comment:
I couldn't repro the ftruncate failure on OS X, but I left the refactoring I did to make it easier to add more checks.

Also, self nit: 'work well with' (will fix before squashing).